quinn/transformations.py

```diff
@@ -87,28 +88,193 @@ def to_snake_case(s: str) -> str:
     :type sort_order: str
+    :param sort_nested: Whether to sort nested structs or not. Defaults to false.
+    :type sort_nested: bool
     :return: A DataFrame with the columns sorted in the chosen order
-    :rtype: pandas.DataFrame
+    :rtype: pyspark.sql.DataFrame
     """
-    sorted_col_names = None
-    if sort_order == "asc":
-        sorted_col_names = sorted(df.columns)
-    elif sort_order == "desc":
-        sorted_col_names = sorted(df.columns, reverse=True)
-    else:
-        raise ValueError(
-            "['asc', 'desc'] are the only valid sort orders and you entered a sort order of '{sort_order}'".format(
-                sort_order=sort_order
+
+    def parse_sort_order(sort_order: str) -> bool:
+        if sort_order not in ["asc", "desc"]:
+            raise ValueError(
+                "['asc', 'desc'] are the only valid sort orders and you entered a sort order of '{sort_order}'".format(
+                    sort_order=sort_order
+                )
             )
-        )
-    return df.select(*sorted_col_names)
+        reverse_lookup = {
+            "asc": False,
+            "desc": True,
+        }
+        return reverse_lookup[sort_order]
+
+    def sort_top_level_cols(df, is_reversed) -> DataFrame:
+        # sort top level columns
+        sorted_col_names = sorted(df.columns, reverse=is_reversed)
+        return df.select(*sorted_col_names)
+
+    def sort_nested_cols(schema, is_reversed, base_field="") -> list[str]:
+        # recursively check nested fields and sort them
+        # https://stackoverflow.com/questions/57821538/how-to-sort-columns-of-nested-structs-alphabetically-in-pyspark
+        # Credits: @pault for logic
+
+        def parse_fields(
+            fields_to_sort: list, parent_struct, is_reversed: bool
+        ) -> list:
+            sorted_fields: list = sorted(
+                fields_to_sort,
+                key=lambda x: x["name"],
+                reverse=is_reversed,
+            )
+
+            results = []
+            for field in sorted_fields:
+                new_struct = StructType([StructField.fromJson(field)])
+                new_base_field = parent_struct.name
+                if base_field:
+                    new_base_field = base_field + "." + new_base_field
+
+                results.extend(
+                    sort_nested_cols(new_struct, is_reversed, base_field=new_base_field)
+                )
+            return results
+
+        def handle_array_type(parent_struct: StructField, is_reversed: bool) -> str:
+            def format_array_selection(
+                elements: list, base_str: str, suffix_str: str
+            ) -> str:
+                element_names = [i.split(".")[-1] for i in elements]
+                array_elements_formatted = [f"x.{i} as {i}" for i in element_names]
+
+                output = (
+                    f"{base_str} {', '.join(array_elements_formatted)} {suffix_str}"
```

Comment:
Very hard to imagine the result of this f-string. May you add few comments here, please?